build/version.go

```diff
@@ -122,11 +122,5 @@ func WithBuildInfo(ctx context.Context, cfg *LogConfig) (context.Context,
 		return nil, fmt.Errorf("unable to decode commit hash: %w", err)
 	}
 
-	// Include the first 3 bytes of the commit hash in the context as an
-	// slog attribute.
-	if len(commitHash) > 3 {
-		commitHash = commitHash[:3]
-	}
-
-	return btclog.WithCtx(ctx, btclog.Hex("rev", commitHash)), nil
+	return btclog.WithCtx(ctx, btclog.Hex3("rev", commitHash)), nil
```

Comment:
I assume this is why we need to update the `btclog` dependency?
Perhaps it would be a good idea to start a `lndlog` or `lnlog` package (not submodule for ease of use) in this project where we can place such new log related helper functions.

Reply:
this was just a nice to have. The main reason for the update is the skipping of quoting strings with newlines as this breaks the spew.DumpS closures

Reply:
but yeah - agreed re the helper package idea! 

(but we still need the btclog up date for this PR if we want to switch to slogs here cause of the spew thing)